core/modules/ccontrol/MergingRequester.cc

```diff
@@ -93,12 +94,16 @@ bool MergingRequester::flush(int bLen, bool last) {
     case MsgState::HEADER_SIZE_WAIT:
         _response->headerSize = static_cast<unsigned char>(_buffer[0]);
         if (!proto::ProtoHeaderWrap::unwrap(_response, _buffer)) {
-            _setError(log::MSG_RESULT_DECODE,
-                std::string("Error decoding proto header for ") + getStateStr(_state));
+             std::ostringstream os;
+            os << "From:" << _wName << "Error decoding proto header for " << getStateStr(_state);
```

Comment:
nitpick: this could still be done with just string concatenation which should be somewhat more efficient that creating ostringstream.